support_for_advertising_use_cases.md

```diff
@@ -529,51 +531,55 @@ This use case is common in sales period. Note that sales period can have very pr
+This can be done programmatically using historical aggregated and/or personal user data, and contextual information.
 
 ## Coupon Management
 
 As an Advertiser I want to be able to add coupons, or any form of targeted discounts to my creatives on specific period of times, potentially for specific products.
 
+# ML Applications
 
-## Recommendation
+Machine Learning Models are very important part of optimal investment in advertising. They are used by many participants in three major areas:
 
-Large e-Commerce websites may have product catalogs with millions of items. They have an interesting use-case. When running ads to promote items from their vast catalogs, which product should they show?
-Today, much of this is mainly powered by data about which items a person has previously browsed.
+## Item recommendations
 
-Typical sources of recommendation would be :
-- Historical Products (compared to historical products browsed): abandoned cart, etc.
-- Similar products (compared to historical products browsed)
-- Complementary Products (compared to historical products browsed)
-- ...
+When there are many items to choose from (products/content/messages/apps) before rendering ad, which one should be shown to which person and in which ad? This produces more useful ads for consumers. Users are less annoyed with ads because they will more likely see items they would like to buy/convert or see less products they already bought/installed or they are not interested to buy/install.
 
-# Training ML Models
+## Bidding logic
+
+Used for estimating efficiency of ad placements. When there are millions publishers to chose from and even more ad placements on those publishers with multiple available ad sizes, which one has highest chances of end conversion for advertisers and which one has highest return on investment? This drives better monetization opportunity for publishers and creates equal chances for big and small publishers to compete with performance results they do deliver for advertising campaign.
+
+## Ad selection
+
+When there are many template ads (style of ad, banner, native, video etc) to choose from, which one should be shown to which person? This produces more attractive ads for people and by increasing overall efficiency it drives publishers revenues.
 
-Ad selection is a very important task. When there are many ads to choose from, which one should be shown to which person? The most effective approach to-date has proven to be using machine learning algorithms to assist with ad selection. This produces more relevant ads for people, and more business outcomes for advertisers. This in turn increases publisher revenues. 
+The most effective approach to-date has proven to be using machine learning algorithms to assist with ad selection, product recommendation and in bidding logic. This produces more usefull ads for people, and more business outcomes for advertisers. This in turn increases publisher revenues.
```

Comment:
I think it's worth splitting this out into two separate ML use-cases.

I would prefer to keep the section that was removed and call that: "Ad Selection".

The newly suggested language seems to refer to a different use-case: Given an ad, what is the optimal format to use to render it? Perhaps we can call this use-case "Format automation". 

Here is my suggestion:

## Ad Selection

Ad selection is a very important task. When there are many ads to choose from, which one should be shown to which person? The most effective approach to-date has proven to be using machine learning algorithms to assist with ad selection. This produces more relevant ads for people, and more business outcomes for advertisers. This in turn increases publisher revenues. 

You also briefly touch upon product recommendation, but that use-case was already covered previously in the section called: "Item recommendations"

## Format automation

Once a particular ad has been selected, there remains the important task of deciding how to render it. There are many "degrees of freedom" to select from:
- Which format? Video? Image? Other?
- What color to use?
- What size of borders?
- Which description / title to use?